src/build_log.cc

```diff
@@ -65,7 +66,7 @@ uint64_t MurmurHash64A(const void* key, size_t len) {
   const uint64_t m = BIG_CONSTANT(0xc6a4a7935bd1e995);
   const int r = 47;
   uint64_t h = seed ^ (len * m);
-  const unsigned char* data = (const unsigned char*)key;
+  const unsigned char* data = static_cast<const unsigned char*>(key);
```

Comment:
Don't static cast a pointer. Reinterpret cast it.

Reply:
Should still be reinterpret_cast for consistancy.

Reply:
I'm not understanding.

key is "const void *".

So reinterpret_cast<const unsigned char *>(const void *) should be fine, shouldn't it?

Reply:
Ah never mind. Thought it was for a different change.

Reply:
I would rather avoid reinterpret_cast whenever possible. When someone changes the type of key, they should get a compile error.

Reply:
How about we change the type right now then? If the parameter is changed to char* it can't be messed up.

Reply:
Either that or the static_cast.

Reply:
Sure. Either way.

I agree that static_cast has the property of ensuring a compiler error if the type being casted from can't be casted with static_cast. That's a beneficial outcome.

Personally I prefer the parameter to this function change type, but I'm not that concerned about it, so whatever way works.